### PR TITLE
feat(swap): handle new events

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/swap/prisma/schema.prisma [hunk 1]
```diff
@@ -100,6 +100,60 @@ model SwapDepositChannelAffiliate {
...
+enum SwapRequestOrigin {
+  DEPOSIT_CHANNEL
+  VAULT
+  INTERNAL
+
+  @@schema("public")
+}
+
+enum SwapRequestType {
+  LEGACY_SWAP
```
Comment (jerryafr) [inline packages/swap/prisma/schema.prisma:112]: 👍

Comment (acdibble) [reply packages/swap/prisma/schema.prisma:112]: Should maybe add legacy ccm too

**Review by zoheb391 — APPROVED** [review]
send it 🫡

File: packages/swap/src/event-handlers/networkDepositFinalised.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    } else if ('gasSwapId' in action && action.gasSwapId !== null) {
+      swapRequestId = action.gasSwapId;
+    } else if ('swapId' in action && action.swapId !== null) {
+      swapRequestId = action.swapId;
+    } else if ('swapRequestId' in action && action.swapRequestId !== null) {
+      swapRequestId = action.swapRequestId;
+    }
+
+    if (!swapRequestId) {
+      logger.warn('No swapRequestId found in networkDepositReceived');
```
Comment (GabrielBuragev) [inline packages/swap/src/event-handlers/networkDepositFinalised.ts:107]: should we maybe crash here?

File: packages/swap/src/event-handlers/refundEgressScheduled.ts [hunk 3]
```diff
@@ -1,50 +1,45 @@
...
+    swapRequestId,
     egressId: [chain, nativeId],
     egressFee,
     amount,
+    asset,
   } = eventArgs.parse(event.args);
 
-  const swap = await prisma.swap.findUniqueOrThrow({
-    where: { nativeId: swapId },
-  });
```
Comment (GabrielBuragev) [inline packages/swap/src/event-handlers/refundEgressScheduled.ts:28]: we don't want to make sure there is an entry in the db anymore?

File: packages/swap/src/event-handlers/swapEgressScheduled.ts [hunk 4]
```diff
@@ -60,52 +23,29 @@ export default async function swapEgressScheduled({
...
     egressId: [chain, nativeId],
-    ...restArgs
+    egressFee,
+    amount,
+    asset,
   } = eventArgs.parse(event.args);
 
-  const swap = await prisma.swap.findUniqueOrThrow({
-    where: { nativeId: swapId },
-  });
```
Comment (GabrielBuragev) [inline packages/swap/src/event-handlers/swapEgressScheduled.ts:70]: same here

**Review by GabrielBuragev — APPROVED** [review]
A few small things but nothing major. Looks clean and very easy to read/understand! Great job!

🧹

**Pull request merged**